# Predicting credit risk for Loans

In [1]:
import pandas as pd

pd.set_option('max_columns', 180)
pd.set_option('max_rows', 50000)
pd.set_option('max_colwidth', 5000)

loans_data = pd.read_csv("loans_2007_2011.csv",low_memory=False)
loans_data.shape

(42538, 52)

In [2]:
loans_data.head(10)

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501  1296599.0     5000.0       5000.0           4975.0   36 months   
1  1077430  1314167.0     2500.0       2500.0           2500.0   60 months   
2  1077175  1313524.0     2400.0       2400.0           2400.0   36 months   
3  1076863  1277178.0    10000.0      10000.0          10000.0   36 months   
4  1075358  1311748.0     3000.0       3000.0           3000.0   60 months   
5  1075269  1311441.0     5000.0       5000.0           5000.0   36 months   
6  1069639  1304742.0     7000.0       7000.0           7000.0   60 months   
7  1072053  1288686.0     3000.0       3000.0           3000.0   36 months   
8  1071795  1306957.0     5600.0       5600.0           5600.0   60 months   
9  1071570  1306721.0     5375.0       5375.0           5350.0   60 months   

  int_rate  installment grade sub_grade                  emp_title emp_length  \
0   10.65%       162.87     B        B2                        NaN  10+ years   
1   15.27%        59.83     C        C4                      Ryder   < 1 year   
2   15.96%        84.33     C        C5                        NaN  10+ years   
3   13.49%       339.31     C        C1        AIR RESOURCES BOARD  10+ years   
4   12.69%        67.79     B        B5   University Medical Group     1 year   
5    7.90%       156.46     A        A4       Veolia Transportaton    3 years   
6   15.96%       170.08     C        C5  Southern Star Photography    8 years   
7   18.64%       109.43     E        E1            MKC Accounting     9 years   
8   21.28%       152.39     F        F2                        NaN    4 years   
9   12.69%       121.45     B        B5                  Starbucks   < 1 year   

  home_ownership  annual_inc verification_status   issue_d  loan_status  \
0           RENT     24000.0            Verified  Dec-2011   Fully Paid   
1           RENT     30000.0     Source Verified  Dec-2011  Charged Off   
2           RENT     12252.0        Not Verified  Dec-2011   Fully Paid   
3           RENT     49200.0     Source Verified  Dec-2011   Fully Paid   
4           RENT     80000.0     Source Verified  Dec-2011      Current   
5           RENT     36000.0     Source Verified  Dec-2011   Fully Paid   
6           RENT     47004.0        Not Verified  Dec-2011   Fully Paid   
7           RENT     48000.0     Source Verified  Dec-2011   Fully Paid   
8            OWN     40000.0     Source Verified  Dec-2011  Charged Off   
9           RENT     15000.0            Verified  Dec-2011  Charged Off   

  pymnt_plan             purpose                                  title  \
0          n         credit_card                               Computer   
1          n                 car                                   bike   
2          n      small_business                   real estate business   
3          n               other                               personel   
4          n               other                               Personal   
5          n             wedding  My wedding loan I promise to pay back   
6          n  debt_consolidation                                   Loan   
7          n                 car                        Car Downpayment   
8          n      small_business   Expand Business & Buy Debt Portfolio   
9          n               other            Building my credit history.   

  zip_code addr_state    dti  delinq_2yrs earliest_cr_line  inq_last_6mths  \
0    860xx         AZ  27.65          0.0         Jan-1985             1.0   
1    309xx         GA   1.00          0.0         Apr-1999             5.0   
2    606xx         IL   8.72          0.0         Nov-2001             2.0   
3    917xx         CA  20.00          0.0         Feb-1996             1.0   
4    972xx         OR  17.94          0.0         Jan-1996             0.0   
5    852xx         AZ  11.20          0.0         Nov-2004             3.0   
6    280xx         NC  23.51          0.0         Jul-2005 

In [3]:
loans_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42538 entries, 0 to 42537
Data columns (total 52 columns):
id                            42538 non-null object
member_id                     42535 non-null float64
loan_amnt                     42535 non-null float64
funded_amnt                   42535 non-null float64
funded_amnt_inv               42535 non-null float64
term                          42535 non-null object
int_rate                      42535 non-null object
installment                   42535 non-null float64
grade                         42535 non-null object
sub_grade                     42535 non-null object
emp_title                     39909 non-null object
emp_length                    41423 non-null object
home_ownership                42535 non-null object
annual_inc                    42531 non-null float64
verification_status           42535 non-null object
issue_d                       42535 non-null object
loan_status                   42535 non-null object
p

### Check for missing values

In [4]:
missing_values = loans_data.isnull().sum()
missing_values = missing_values*100/loans_data.shape[0]
missing_values.sort_values(ascending=False)

emp_title                     6.180356
pub_rec_bankruptcies          3.215948
emp_length                    2.621186
chargeoff_within_12_mths      0.347924
collections_12_mths_ex_med    0.347924
tax_liens                     0.253891
revol_util                    0.218628
last_pymnt_d                  0.202172
total_acc                     0.075227
delinq_2yrs                   0.075227
earliest_cr_line              0.075227
open_acc                      0.075227
pub_rec                       0.075227
inq_last_6mths                0.075227
acc_now_delinq                0.075227
delinq_amnt                   0.075227
title                         0.037613
annual_inc                    0.016456
last_credit_pull_d            0.016456
total_rec_int                 0.007053
pymnt_plan                    0.007053
issue_d                       0.007053
verification_status           0.007053
home_ownership                0.007053
sub_grade                     0.007053
grade                    

None of the columns appear to have more than ~6.2% missing values, hence we can proceed with the dataset without the initial need to drop any columns based solely off missing values.

The [data dictionary](https://docs.google.com/spreadsheets/d/191B2yJ4H1ZPXq0_ByhUgWMFZOYem5jFz0Y3by_7YBY4/edit#gid=1252101565) for the dataset was acquired from Kaggle's website uploaded by Wendy Kan. Here is the [link](https://www.kaggle.com/wendykan/lending-club-loan-data) to her files.

Based off the data dictionary, it is clear that there is a redundancy in the columns and data available post issuance of the loan amount to the borrowers. The data made available after the loan has been issued cannot be included as part of the training set because it wouldn't be available when the model is deployed on new incoming data.

The following columns can be dropped due to the above listed reasons:

| Column_name | Description |
| ----------- | ----------- |
|"id" | Lending Club assigned number for loan listing. |
|"member_id" | Borrower id. |
|"funded_amnt" | Won't be available during prediction. |
|"funded_amnt_inv" | Won't be available during prediction. |
|"grade" | Lending Club assigned loan grade. Redundant with interest rate. |
|"sub_grade" | Lending Club assigned loan subgrade. Redundant with interest rate. |
|"emp_title" | Borrower employment status during application phase. |
|"issue_d" | Loan issue date. Won't be available during prediction. |
|"zip_code" | Zip code of borrower |
|"out_prncp" | Outstanding principal amount remaining. Won't be available during prediction. |
|"out_prncp_inv" | Outstanding principal amount remaining to investors. Won't be available during prediction. |
|"total_pymnt" | Payments received to date for total amount funded. Won't be available during prediction. |
|"total_pymnt_inv" | Payments received to date for portion of total amount funded by investors. Won't be available during prediction. |
|"total_rec_prncp" | Principal received to date. Won't be available during prediction. |
|total_rec_int | Won't be available during prediction.|
|total_rec_late_fee | Won't be available during prediction. |
|recoveries | Won't be available during prediction. |
|collection_recovery_fee | Won't be available during prediction. |
|last_pymnt_d | Won't be available during prediction. |
|last_pymnt_amnt | Won't be available during prediction. |

In [5]:
loans_data = loans_data.drop(["id", "member_id", "funded_amnt", "funded_amnt_inv", "grade", "sub_grade", "emp_title", "issue_d",
                             "zip_code", "out_prncp", "out_prncp_inv", "total_pymnt", "total_pymnt_inv", "total_rec_prncp",
                             "total_rec_int", "total_rec_late_fee", "recoveries", "collection_recovery_fee", "last_pymnt_d", "last_pymnt_amnt"], axis=1)

In [6]:
loans_data.head(10)

loan_amnt        term int_rate  installment emp_length home_ownership  \
0     5000.0   36 months   10.65%       162.87  10+ years           RENT   
1     2500.0   60 months   15.27%        59.83   < 1 year           RENT   
2     2400.0   36 months   15.96%        84.33  10+ years           RENT   
3    10000.0   36 months   13.49%       339.31  10+ years           RENT   
4     3000.0   60 months   12.69%        67.79     1 year           RENT   
5     5000.0   36 months    7.90%       156.46    3 years           RENT   
6     7000.0   60 months   15.96%       170.08    8 years           RENT   
7     3000.0   36 months   18.64%       109.43    9 years           RENT   
8     5600.0   60 months   21.28%       152.39    4 years            OWN   
9     5375.0   60 months   12.69%       121.45   < 1 year           RENT   

   annual_inc verification_status  loan_status pymnt_plan             purpose  \
0     24000.0            Verified   Fully Paid          n         credit_card   
1     30000.0     Source Verified  Charged Off          n                 car   
2     12252.0        Not Verified   Fully Paid          n      small_business   
3     49200.0     Source Verified   Fully Paid          n               other   
4     80000.0     Source Verified      Current          n               other   
5     36000.0     Source Verified   Fully Paid          n             wedding   
6     47004.0        Not Verified   Fully Paid          n  debt_consolidation   
7     48000.0     Source Verified   Fully Paid          n                 car   
8     40000.0     Source Verified  Charged Off          n      small_business   
9     15000.0            Verified  Charged Off          n               other   

                                   title addr_state    dti  delinq_2yrs  \
0                               Computer         AZ  27.65          0.0   
1                                   bike         GA   1.00          0.0   
2                   real estate business         IL   8.72          0.0   
3                               personel         CA  20.00          0.0   
4                               Personal         OR  17.94          0.0   
5  My wedding loan I promise to pay back         AZ  11.20          0.0   
6                                   Loan         NC  23.51          0.0   
7                        Car Downpayment         CA   5.35          0.0   
8   Expand Business & Buy Debt Portfolio         CA   5.55          0.0   
9            Building my credit history.         TX  18.08          0.0   

  earliest_cr_line  inq_last_6mths  open_acc  pub_rec  revol_bal revol_util  \
0         Jan-1985             1.0       3.0      0.0    13648.0      83.7%   
1         Apr-1999             5.0       3.0      0.0     1687.0       9.4%   
2         Nov-2001             2.0       2.0      0.0     2956.0      98.5%   
3         Feb-1996             1.0      10.0      0.0     5598.0        21%   
4         Jan-1996             0.0      15.0      0.0    27783.0      53.9%   
5         Nov-2004             3.0       9.0      0.0     7963.0      28.3%   
6         Jul-2005             1.0       7.0      0.0    17726.0      85.6%   
7         Jan-2007             2.0       4.0      0.0     8221.0      87.5%   
8         Apr-2004             2.0      11.0      0.0     5210.0      32.6%   
9         Sep-2004             0.0       2.0      0.0     9279.0      36.5%   

   total_acc initial_list_status last_credit_pull_d  \
0        9.0                   f           Jun-2016   
1        4.0                   f           Sep-2013   
2       10.0                   f           Jun-2016   
3       37.0                   f           Apr-2016   
4       38.0                   f           Jun-2016   
5       12.0                   f           Jan-2016   
6       11.0                   f           May-2016   
7        4.0                   f           Dec-2014   
8       13.0                   f           Aug-2012   
9        3.0                

### Target variable

The choice of target variable is key to our analysis as we are looking to predict, if based on a given set of features, a borrower is a good for the loan amount.

The only variable that fits the bill is "loan_status" as it describes the activity of the borrower w.r.t the loan i.e. it directly describes if a loan was paid off on time, had delayed payments, or was defaulted on the borrower.

In [7]:
loans_data['loan_status'].value_counts(normalize=True)*100

Fully Paid                                             77.902903
Charged Off                                            13.245562
Does not meet the credit policy. Status:Fully Paid      4.673798
Current                                                 2.259316
Does not meet the credit policy. Status:Charged Off     1.789115
Late (31-120 days)                                      0.056424
In Grace Period                                         0.047020
Late (16-30 days)                                       0.018808
Default                                                 0.007053
Name: loan_status, dtype: float64

~78% of the loans offered via Lending Club have been paid off in full. Only ~0.01% of the loans are defaulted upon.

The meaning behind each status message was learnt through their [webpage](https://help.lendingclub.com/hc/en-us/articles/215488038-What-do-the-different-Note-statuses-mean-). 

| Loan Status |	Meaning |
| ----------- | ------- |
| Fully Paid	|	Loan has been fully paid off. |
| Charged Off	|	Loan for which there is no longer a reasonable expectation of further payments. |
| Does not meet the credit policy. Status:Fully Paid	|	The loan was paid off, application wouldn't be approved on to the marketplace in the present. |
| Does not meet the credit policy. Status:Charged Off | The loan was charged off, application wouldn't be approved on to the marketplace in the present. |
| In Grace Period	|	The loan is past due but still in the grace period of 15 days. |
| Late (16-30 days)	|	Loan hasn't been paid in 16 to 30 days (late on the current payment). |
| Late (31-120 days)	|	Loan hasn't been paid in 31 to 120 days (late on the current payment). |
| Current	|	Loan is up to date on current payments. |
| Default	|	Loan is defaulted on and no payment has been made for more than 121 days. 



Since the risk prediction for a potential new loan can only be a "Yes" or "No", the problem is essentially a binary classification. The status messages that best reflect the outcomes desired are "Fully Paid" and "Charged Off", both of these messages imply that there is no expectation for any future payments no matter how marginal the chances.

We adopt a binary notation moving forward, "Fully Paid" equals 1 and "Charged Off" is equal to 0.

In [8]:
loans_data = loans_data[(loans_data['loan_status'] == "Fully Paid") | (loans_data['loan_status'] == "Charged Off")]

status_replace = {
    "loan_status" : {
        "Fully Paid": 1,
        "Charged Off": 0,
    }
}

loans_data = loans_data.replace(status_replace)

We can also drop columns with a lack of unique values. These columns don't change across the dataset and hence would not add any valuable information to our modeling.

In [9]:
original_columns = loans_data.columns
drop_columns = []
for col in original_columns:
    col_series = loans_data[col].dropna().unique()
    if len(col_series) == 1:
        drop_columns.append(col)

print(drop_columns)
clean_loans_data = loans_data.drop(drop_columns, axis=1)

['pymnt_plan', 'initial_list_status', 'collections_12_mths_ex_med', 'policy_code', 'application_type', 'acc_now_delinq', 'chargeoff_within_12_mths', 'delinq_amnt', 'tax_liens']


In [10]:
clean_loans_data.isnull().sum()

loan_amnt                  0
term                       0
int_rate                   0
installment                0
emp_length              1036
home_ownership             0
annual_inc                 0
verification_status        0
loan_status                0
purpose                    0
title                     11
addr_state                 0
dti                        0
delinq_2yrs                0
earliest_cr_line           0
inq_last_6mths             0
open_acc                   0
pub_rec                    0
revol_bal                  0
revol_util                50
total_acc                  0
last_credit_pull_d         2
pub_rec_bankruptcies     697
dtype: int64

Looking at the null values in the columns we have retained moving forward, 2 columns stand out from the rest.

"emp_length" which is the number of years the borrower has been employed at a given job, it is regarded as a key identifier of the financial stability of the borrower and must be retained.

In [11]:
print(clean_loans_data.pub_rec_bankruptcies.value_counts(normalize=True, dropna=False))

0.0    0.939438
1.0    0.042456
NaN    0.017978
2.0    0.000129
Name: pub_rec_bankruptcies, dtype: float64


Close to ~94% of the column is filled by 0, making this column another candidate to be dropped completely as it adds no value to our predictive analysis.

In [12]:
clean_loans_data = clean_loans_data.drop("pub_rec_bankruptcies", axis=1)
clean_loans_data = clean_loans_data.dropna(axis=0)

### Cleaning column dtypes

In [13]:
object_columns_df = clean_loans_data.select_dtypes(include=["object"])
print(object_columns_df.iloc[0])

term                     36 months
int_rate                    10.65%
emp_length               10+ years
home_ownership                RENT
verification_status       Verified
purpose                credit_card
title                     Computer
addr_state                      AZ
earliest_cr_line          Jan-1985
revol_util                   83.7%
last_credit_pull_d        Jun-2016
Name: 0, dtype: object


In [14]:
cols = ['home_ownership', 'verification_status', 'emp_length', 'term', 'addr_state', 'purpose']

for c in cols:
    print("\n ___New Column___ \n")
    print(clean_loans_data[c].value_counts())
    



 ___New Column___ 

RENT        18112
MORTGAGE    16686
OWN          2778
OTHER          96
NONE            3
Name: home_ownership, dtype: int64

 ___New Column___ 

Not Verified       16281
Verified           11856
Source Verified     9538
Name: verification_status, dtype: int64

 ___New Column___ 

10+ years    8545
< 1 year     4513
2 years      4303
3 years      4022
4 years      3353
5 years      3202
1 year       3176
6 years      2177
7 years      1714
8 years      1442
9 years      1228
Name: emp_length, dtype: int64

 ___New Column___ 

 36 months    28234
 60 months     9441
Name: term, dtype: int64

 ___New Column___ 

CA    6776
NY    3614
FL    2704
TX    2613
NJ    1776
IL    1447
PA    1442
VA    1347
GA    1323
MA    1272
OH    1149
MD    1008
AZ     807
WA     788
CO     748
NC     729
CT     711
MI     678
MO     648
MN     581
NV     466
SC     454
WI     427
OR     422
AL     420
LA     420
KY     311
OK     285
KS     249
UT     249
AR     229
DC     209
RI     19

In [15]:
clean_loans_data["title"].value_counts().head(10)

Debt Consolidation           2068
Debt Consolidation Loan      1599
Personal Loan                 624
Consolidation                 488
debt consolidation            466
Credit Card Consolidation     345
Home Improvement              336
Debt consolidation            314
Small Business Loan           298
Credit Card Loan              294
Name: title, dtype: int64

The "title" column is a string dtype column with the reasoning behind the borrowers application for a loan. There are more than 18000+ reasons provided by the borrowers for their loan requests.

Debt consolidation appears  to be a popular reason for requesting loans.

The "emp_length" column presents data as a string but can be converted to a numerical column.

In [16]:
mapping_dict = {
    "emp_length": {
        "10+ years": 10,
        "9 years": 9,
        "8 years": 8,
        "7 years": 7,
        "6 years": 6,
        "5 years": 5,
        "4 years": 4,
        "3 years": 3,
        "2 years": 2,
        "1 year": 1,
        "< 1 year": 0,
        "n/a": 0
    }
}
clean_loans_data = clean_loans_data.drop(["last_credit_pull_d", "earliest_cr_line", "addr_state", "title"], axis=1)
clean_loans_data["int_rate"] = clean_loans_data["int_rate"].str.rstrip("%").astype("float")
clean_loans_data["revol_util"] = clean_loans_data["revol_util"].str.rstrip("%").astype("float")
clean_loans_data = clean_loans_data.replace(mapping_dict)

### Dummy coding categorical variables

In [17]:
cat_columns = ["home_ownership", "verification_status", "purpose", "term"]

dummy_df = pd.get_dummies(clean_loans_data[cat_columns])
clean_loans_data = pd.concat([clean_loans_data, dummy_df], axis=1)
clean_loans_data = clean_loans_data.drop(cat_columns, axis=1)

## Modeling, Analysis, and Predicting

In [18]:
import numpy as np
from sklearn.linear_model import LogisticRegression

logit_reg = LogisticRegression(solver='liblinear')
cols = clean_loans_data.columns
train_cols = cols.drop("loan_status")
features = clean_loans_data[train_cols]
target = clean_loans_data["loan_status"]
logit_reg.fit(features, target)
predictions = logit_reg.predict(features)

In [19]:
from sklearn.model_selection import cross_val_predict
logit_reg = LogisticRegression(solver='liblinear')
predictions = cross_val_predict(logit_reg, features, target, cv=5)
predictions = pd.Series(predictions)

# False positives.
fp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 0)
fp = len(predictions[fp_filter])

# True positives.
tp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 1)
tp = len(predictions[tp_filter])

# False negatives.
fn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 1)
fn = len(predictions[fn_filter])

# True negatives
tn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 0)
tn = len(predictions[tn_filter])
# Rates
tpr = tp  / (tp + fn)
fpr = fp  / (fp + tn)
print(tpr)
print(fpr)

0.9984268484530676
0.9986179664363277


An issue that must be dealt with in our analysis is the class imbalance between the "Fully Paid" and "Charged Off" labels, their ratio is almost 6:1 i.e. for every one loan that is "Charged Off" there are 6 loans fully paid.

We can balance by weighting the dataset appropriately.

In [23]:
logit_reg = LogisticRegression(class_weight="balanced",solver='liblinear')
predictions = cross_val_predict(logit_reg, features, target, cv=5)
predictions = pd.Series(predictions)

# False positives.
fp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 0)
fp = len(predictions[fp_filter])

# True positives.
tp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 1)
tp = len(predictions[tp_filter])

# False negatives.
fn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 1)
fn = len(predictions[fn_filter])

# True negatives
tn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 0)
tn = len(predictions[tn_filter])

# Rates
tpr = tp / (tp + fn)
fpr = fp / (fp + tn)

print(tpr)
print(fpr)

0.6407315154693235
0.6254689042448174


In [24]:
weights = {
    0: 10,
    1: 1
}

logit_reg = LogisticRegression(class_weight=weights,solver='liblinear')
predictions = cross_val_predict(logit_reg, features, target, cv=5)
predictions = pd.Series(predictions)

# False positives.
fp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 0)
fp = len(predictions[fp_filter])

# True positives.
tp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 1)
tp = len(predictions[tp_filter])

# False negatives.
fn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 1)
fn = len(predictions[fn_filter])

# True negatives
tn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 0)
tn = len(predictions[tn_filter])

# Rates
tpr = tp / (tp + fn)
fpr = fp / (fp + tn)

print(tpr)
print(fpr)

0.20755768222338752
0.2073050345508391


In [29]:
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier(class_weight="balanced", random_state=1)
predictions = cross_val_predict(rf, features, target, cv=5)
predictions = pd.Series(predictions)

# False positives.
fp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 0)
fp = len(predictions[fp_filter])

# True positives.`
tp_filter = (predictions == 1) & (clean_loans_data["loan_status"] == 1)
tp = len(predictions[tp_filter])

# False negatives.
fn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 1)
fn = len(predictions[fn_filter])

# True negatives
tn_filter = (predictions == 0) & (clean_loans_data["loan_status"] == 0)
tn = len(predictions[tn_filter])

# Rates
tpr = tp / (tp + fn)
fpr = fp / (fp + tn)

print(tpr)
print(fpr)

C:\Users\vigne\Anaconda3\lib\site-packages\sklearn\ensemble\forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.
  "10 in version 0.20 to 100 in 0.22.", FutureWarning)
C:\Users\vigne\Anaconda3\lib\site-packages\sklearn\ensemble\forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.
  "10 in version 0.20 to 100 in 0.22.", FutureWarning)
C:\Users\vigne\Anaconda3\lib\site-packages\sklearn\ensemble\forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.
  "10 in version 0.20 to 100 in 0.22.", FutureWarning)
C:\Users\vigne\Anaconda3\lib\site-packages\sklearn\ensemble\forest.py:245: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.
  "10 in version 0.20 to 100 in 0.22.", FutureWarning)
C:\Users\vigne\Anaconda3\lib\site-packages\sklearn\ensemble\forest.py:245: F

0.9665049816465653
0.9684106614017769
